src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
```

Comment:
It looks a little bit complicated, you can try smth like this instead
 const motherNames = people.map((el)=>el.mother);
  
 const women = withChildren
    ? people.filter((p) => p.sex === 'f' && motherNames.includes(p.name))
    : people.filter((p) => p.sex === 'f');